jsign: init at 7.2

JohnAZoidberg opened the pull request:
jsign is a nice tool to sign PE (Windows, UEFI) binaries on Linux. I use it with codesigning certificate on a yubikey.
For yubikey `services.pcscd.enable` needs to be enabled.

It works with one of two commands:

```
jsign --storetype YUBIKEY --storepass 123456 --tsaurl http://ts.ssl.com --tsmode RFC3161 foo.exe                                                                                                                    
jsign --storetype PIV --storepass 123456 --tsaurl http://ts.ssl.com --tsmode RFC3161 foo.exe                                                                                                                        
```

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

JohnAZoidberg committed:
jsign: init at 7.2

Signed-off-by: Daniel Schaefer <git@danielschaefer.me>

Tested with yubikey (must have `services.pcscd.enable = true;`):

> jsign --storetype PIV --storepass FOO --tsaurl http://ts.ssl.com --tsmode RFC3161 foo.exe
> jsign --storetype YUBIKEY --storepass FOO --tsaurl http://ts.ssl.com --tsmode RFC3161 foo.exe

Signed-off-by: Daniel Schaefer <git@danielschaefer.me>

acid-bong reviewed (commented):
Moin. A couple of nitpicks

acid-bong commented on pkgs/by-name/js/jsign/package.nix:
```diff
@@ -0,0 +1,72 @@
+          opensc
+        ]
+      } \
+      --add-flags "-Dsun.security.smartcardio.library=${lib.getLib pcsclite.lib}/lib/libpcsclite.so.1 -jar $out/share/jsign.jar"
```
```suggestion
      --add-flags "-Dsun.security.smartcardio.library=${lib.getLib pcsclite}/lib/libpcsclite.so.1 -jar $out/share/jsign.jar"
```
That attr should've been removed

acid-bong reviewed (commented):
Also squash everything, new content is supposed to be added in one step (each!)

JohnAZoidberg commented:
> Also squash everything, new content is supposed to be added in one step (each!)

ok all done

JohnAZoidberg commented:
> You're commit message is wrong btw :p should be init at 7.2

ahh thanks

> Also if the package doesn't need variable expansion at runtime then makeBinaryWrapper is preferred over makeWrapper.
Please also remove meta with lib usage.

It doesn't, thanks, updated

> The Darwin platforms should be marked broken (with meta.broken = stdenv.hostPlatform.isDarwin;) for now and probably be fixed later.

Added

0x4A6F reviewed (approved):
Builds and runs on {aarch64,x86_64}-linux.
